packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,268 @@
+    }
+  }
+
+  /**
+   * Method that gets called whenever the menu changes.
+   * @protected
+   */
+  protected menuChanged(): void {
+    if (this.$fastController.isConnected && this.menu && this.menu.length) {
+      this._menu = this.menu[0] as MenuList;
+    }
+  }
+
+  /**
+   * Method that gets called whenever the trigger changes.
+   * @protected
+   * @param {HTMLElement[] | void} prev - The previous value of the trigger.
+   * @param {HTMLElement[]} next - The new value of the trigger.
+   */
+  protected triggerChanged(prev: HTMLElement[] | void, next: HTMLElement[]): void {
+    if (prev !== next) {
+      if (Array.isArray(prev) && prev.length) {
+        this.cleanupTriggerEventListeners(prev[0]);
+      }
+
+      if (next && next.length) {
+        this._trigger = next[0];
+        this._trigger.setAttribute('aria-haspopup', 'true');
+        this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+        this._trigger.addEventListener('keydown', this.handleTriggerKeydown);
+        this._trigger.addEventListener('click', this.toggleMenu);
+
+        if (this.openOnHover) {
+          this._trigger.addEventListener('mouseover', this.toggleMenu);
+        }
+      }
+    }
+  }
+
+  /**
+   * Method that gets called whenever the expanded state changes.
+   * @protected
+   */
+  protected expandedChanged(oldValue: boolean, newValue: boolean): void {
+    if (this.$fastController.isConnected && this._trigger) {
+      this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+    }
+    Updates.enqueue(() => {
+      this.setPositioning();
+      this.focus();
+    });
+    this.$emit('expandedChanged', { expanded: newValue });
+  }
+
+  /**
+   * Handles document click events to close the menu when a click occurs outside of the menu or the trigger.
+   * @param {Event} e - The event triggered on document click.
+   * @private
+   */
+  private handleDocumentClick = (e: any) => {
+    if (e && !e.composedPath().includes(this._menu) && !e.composedPath().includes(this._trigger)) {
+      this.closeMenu();
+    }
+  };
+
+  /**
+   * Handle keyboard interaction for the menu.
+   *
+   * @param e - the keyboard event
+   * @internal
+   */
+  public handleMenuKeydown(e: KeyboardEvent): boolean | void {
+    const key = e.key;
+
+    switch (key) {
+      case keyTab: {
+        if (this.expanded) {
+          this.closeMenu();
+        }
+        break;
+      }
+      case keyEscape: {
+        if (this.expanded) {
+          e.preventDefault();
+          this.closeMenu();
+          this._trigger?.focus();
+        }
+        break;
+      }
+    }
+  }
+
+  /**
+   * Handles keyboard interaction for the trigger.
+   * @param {KeyboardEvent} e - the keyboard event
+   * @public
+   */
+  public handleTriggerKeydown(e: KeyboardEvent): void {
+    const key = e.key;
+    switch (key) {
+      case keySpace:
+      case keyEnter: {
+        e.preventDefault();
+        this.toggleMenu;
+        break;
+      }
+    }
```

Comment:
We my want to return true as default here to allow the event to continue bubbling.